gateway/mw_url_rewrite.go

```diff
@@ -384,7 +386,9 @@ func (m *URLRewriteMiddleware) ProcessRequest(w http.ResponseWriter, r *http.Req
 	if err != nil {
 		log.Error("URL Rewrite failed, could not parse: ", p)
 	} else {
-		r.URL = newURL
+		//Setting new path here breaks request middleware
```

Comment:
No need to `else` statement

Reply:
We don't call `ctxSetURLRewriteTarget` when `err!=nil` and we don't return inside `err` check, else is required here.

Reply:
yep I missed.